docs/deployment-integration.md

```diff
@@ -0,0 +1,312 @@
+ their BTC across too many networks (recommended: `3-5`)
+<!-- TODO: this should not be here, we will have a global limit on fps -->
+* `rollup_finality_contract_address`: Babylon Genesis address of the deployed Rollup BSN contract (`bbn1...` format)
+
+> **Note**: The above operation requires governance in permissioned
+> registration networks.
+
+## Contract Maintenance
+<img width="1848" height="677" alt="adminRole" src="https://github.com/user-attachments/assets/9b942068-3d72-4ae8-a47e-79894427204e" />
+
+The Rollup BSN contract exposes `admin-only` functions for ongoing management. These allow you to clean up
+ old data, update roles, and adjust configuration without redeploying the contract.
+
+Each operation is executed with:
+
+```shell
+babylond tx wasm execute <CONTRACT_ADDRESS> '<MSG>' \
+  --from $ADMIN_KEY \
+  --chain-id $BBN_CHAIN_ID \
+  --keyring-backend test \
+  --gas auto \
+  --gas-adjustment 1.3 \
+  --fees 1000000ubbn \
+  -y
+```
+
+### Modifying the Contract Administrator
+```shell
+MSG={
+  "update_admin": {
+    "admin": <NEW_ADMIN_ADDRESS>
+  }
+}
+```
+Transfers contract admin rights to a new Babylon address. Typically used when rotating keys or migrating
+ ownership to another entity
+
+### Data Pruning
+```shell
+MSG={
+  "prune_data": {
+    "rollup_height": <BLOCK_HEIGHT>,
+    "max_signatures_to_prune": <MAX_SIGS>, # optional
+    "max_pub_rand_values_to_prune": <MAX_PUB_RAND> # optional
+  }
+}
+```
+Removes all finality signatures and public randomness for rollup blocks with height less than or
+ equal to the specified value. Useful for reducing on-chain storage as the chain grows. Optional
+  fields control pruning batch size to avoid gas exhaustion.
+
+See [PRUNING.m](./PRUNING.md) for detailed guidelines on selecting safe pruning heights and best
+ practices
+
+### Modifying the Finality Providers Allow-List
+```shell
+MSG={
+  "add_to_allowlist": {
+    "fp_pubkey_hex_list": [
+      <FP_PUBKEY>
+    ]
+  }
+}
+```
+```shell
+MSG={
+  "remove_from_allowlist": {
+    "fp_pubkey_hex_list": [
+      <FP_PUBKEY>
+    ]
+  }
+}
+```
+Adds or removes Finality Providers from the allowlist. Use this to onboard new providers or revoke
+ access from inactive or misbehaving ones
+
+### Admin Queries
+In addition to executing messages, admins can query contract state to inspect configuration and
+ permissions. Use:
+
+```shell
+babylond query wasm contract-state smart <CONTRACT_ADDRESS> '<QUERY_MSG>' \
+  --node $NODE_RPC \
+  --output json
+```
+**Available admin queries:**
+```shell
+QUERY_MSG={
+  "admin": {}
+}
+```
+Returns the current contract admin address
+
+```shell
+QUERY_MSG={
+  "config": {}
+}
+```
+Returns the contract configuration, including BSN ID, rate limits, and other parameters
+
+```shell
+QUERY_MSG={
+  "allowed_finality_providers": {}
+}
+```
+Returns a list of BTC public keys (in hex format) currently allowlisted to submit data
+
+## External Integration
```

Comment:
I'm not really sure as to why we need this section. Even if we do, it surely does not need to be so technical. This is an operator's document on how to maintain a rollup BSN contract. The technical details of the messages that finality providers or the queries that the gadgets are doing are not really relevant.

Reply:
Let's remove it. But we should have dedicated doc explaining not how instantiate and deploy but also how to interact. What queries exists and what messages you can send